runtime/binding-tcp/src/main/java/io/aklivity/zilla/runtime/binding/tcp/internal/stream/TcpClientRouter.java

```diff
@@ -85,7 +88,11 @@ public InetSocketAddress resolve(
 
         try
         {
-            if (beginEx == null)
+            if (capacity.get() <= 0)
+            {
+                //No capacity
+            }
```

Comment:
Label the try `resolve:` and then `break resolve;` if no capacity (instead of empty if-block).